Split ARIA props not prohibited from ARIA props permitted (#2342)
auto-wcag/auto-wcag: _rules/aria-global-property-not-prohibited-kb1m8s.md

### Related rules

- [ARIA state or property is permitted](https://www.w3.org/WAI/standards-guidelines/act/rules/5c01ea/)
- [ARIA state or property has valid value](https://www.w3.org/WAI/standards-guidelines/act/rules/6a7281/)

### Bibliography

- [Understanding Success Criterion 1.3.1: Info and Relationships](https://www.w3.org/WAI/WCAG22/Understanding/info-and-relationships.html)
- [Understanding Success Criterion 4.1.2: Name, Role, Value](https://www.w3.org/WAI/WCAG22/Understanding/name-role-value.html)
- [WAI-ARIA 1.2, Prohibited States and Properties](https://www.w3.org/TR/wai-aria/#prohibitedattributes)
- [WAI-ARIA 1.2, Global States and Properties](https://www.w3.org/TR/wai-aria-1.2/#global_states)
- [ARIA5: Using WAI-ARIA state and property attributes to expose the state of a user interface component](https://www.w3.org/WAI/WCAG22/Techniques/aria/ARIA5)

> ## Test Cases

Author: ```suggestion
## Examples
```

### Passed

#### Passed Example 1

This generic `div` element is allowed the global `aria-live` property.

```html
<div aria-live="polite">I like Bananas</div>
```

#### Passed Example 2

> This `a` element is allowed the `aria-label` property.

Reviewer: Given that the expectation is about the role of the owning element:
```suggestion
This `a` element with an [implicit role][] of `link` is allowed the `aria-label` property.
```

---
id: kb1m8s
name: ARIA global properties not used where prohibited
rules_format: 1.1
rule_type: atomic
description: |
  This rule checks that global WAI-ARIA properties are not on elements whose role prohibits them.
accessibility_requirements:
  aria12:prohibitedattributes:
    title: ARIA 1.2, 5.2.5 Prohibited States and Properties
    forConformance: true
    failed: not satisfied
    passed: satisfied
    inapplicable: satisfied
  wcag-technique:ARIA5: # Using WAI-ARIA state and property attributes to expose the state of a user interface component
    forConformance: false
    failed: not satisfied
    passed: further testing needed
    inapplicable: further testing needed
  wcag20:1.3.1: # Info and Relationships (A)
    secondary: This success criterion is **less strict** than this rule. This is because the rule does not ignore irrelevant ARIA properties. Some of the failed examples satisfy this success criterion.
input_aspects:
  - Accessibility Tree
  - CSS styling
  - DOM Tree
acknowledgments:
  authors:
    - Wilco Fiers
  previous_authors:
    - Anne Thyme Nørregaard
    - Jean-Yves Moyen
  assets:

>     - JFK's "We Choose the Moon" speech excerpt is courtesy of NASA.

Author: ```suggestion
```

---

> ## Applicability

Reviewer 2: Maybe a nitpick: I would feel better if the word "attribute" was mentioned here (in the applicability).   This would match other rules such as https://www.w3.org/WAI/standards-guidelines/act/rules/5f99a7/ , https://www.w3.org/WAI/standards-guidelines/act/rules/6a7281/ , https://www.w3.org/WAI/standards-guidelines/act/rules/674b10/ .

Reviewer 2: This comment also applies to the other rule.

This rule applies to any [WAI-ARIA global state or property][global] that is specified on an [HTML or SVG element][namespaced element] that is [included in the accessibility tree][].

## Expectation

> No test target is [prohibited][] on the [semantic role][] of the element on which it is specified.

Author: I'm not sure about this one. Is it "on" or "for"?

```suggestion
The test target is not [prohibited][] for the [semantic role][] of the element on which it is specified.
```

> No test target is [prohibited][] on the [semantic role][] of the element on which it is specified.

Reviewer 3: ```suggestion
The test target is not [prohibited][] on the [semantic role][] of the element on which it is specified.
```

### Related rules

- [ARIA state or property is permitted](https://www.w3.org/WAI/standards-guidelines/act/rules/5c01ea/)
- [ARIA state or property has valid value](https://www.w3.org/WAI/standards-guidelines/act/rules/6a7281/)

### Bibliography

- [Understanding Success Criterion 1.3.1: Info and Relationships](https://www.w3.org/WAI/WCAG22/Understanding/info-and-relationships.html)
- [Understanding Success Criterion 4.1.2: Name, Role, Value](https://www.w3.org/WAI/WCAG22/Understanding/name-role-value.html)
- [WAI-ARIA 1.2, Prohibited States and Properties](https://www.w3.org/TR/wai-aria/#prohibitedattributes)
- [WAI-ARIA 1.2, Global States and Properties](https://www.w3.org/TR/wai-aria-1.2/#global_states)
- [ARIA5: Using WAI-ARIA state and property attributes to expose the state of a user interface component](https://www.w3.org/WAI/WCAG22/Techniques/aria/ARIA5)

## Test Cases

### Passed

#### Passed Example 1

This generic `div` element is allowed the global `aria-live` property.

```html
<div aria-live="polite">I like Bananas</div>
```

#### Passed Example 2

> This `a` element allowed the `aria-label` property.

Reviewer 3: ```suggestion
This `a` element is allowed the `aria-label` property.
```

```html
<a href="#" aria-label="Previously 100, now 1 euro"> <s>€100</s> / <b>€1</b> </a>
```

#### Passed Example 3

> This `div` element allowed the `aria-braillelabel` property because its [explicit role][] of `heading` does not prohibit this.

Reviewer 3: ```suggestion
This `div` element is allowed the `aria-braillelabel` property because its [explicit role][] of `heading` does not prohibit this.
```

```html
<div role="heading" aria-braillelabel="I like bananas">
	I ❤️ Bananas
</div>
```

### Failed

#### Failed Example 1

The `aria-label` property is [prohibited][] for an element with a `generic` role.

```html
<div aria-label="Previously 100, now 1 euro"><s>€100</s> / <b>€1</b></div>
```

#### Failed Example 2

The `aria-labelledby` property is [prohibited][] for an element with a `paragraph` role.

```html
<h1 id="bananas">I like bananas</h1>
<p aria-labelledby="Bananas">🧑 ❤️ 🍌🍌</p>
```

#### Failed Example 3

The `aria-braillelabel` property is [prohibited][] for an element with a `button` role.

```html
<p aria-braillelabel="I love Bananas">I ❤️ Bananas</p>
```

#### Failed Example 4

The `aria-roledescription` property is [prohibited][] for an element with a `generic` role.

```html
<div aria-roledescription="Banana text">I like bananas</div>
```

#### Failed Example 5

The `aria-brailleroledescription` property is [prohibited][] for an element with a `none` role.

```html
<h1 role="none" aria-brailleroledescription="Banana text">I like bananas</h1>

> ```

Reviewer 4: do you think this could be replaced by role="presentation"? it would be clearer because I had to search that `role="none"` is just a synonym to that

Author: Discussed on the call. "none" tends to be the preferred name for this these days.
